docs/docs/guides/atom-effects.md

```diff
@@ -239,6 +241,80 @@ const currentUserIDState = atom({
 });
 ```
 
+### Asynchronous
+
+If your persisted data needs to be `await`ed, you can either create an `asynchronous` function and call that or use a `promise` in the setSelf function 
+
+- Using the promise approach you'll be able to wrap the components inside of the `<RecoilRoot/>` with a `<Suspense/>` component to show a fallback state while waiting for `Recoil` to load the persisted values.
```

Comment:
Let's not call this the "promise approach", as `Promise`s may be used in other approaches as well.  Maybe "`Synchronously call setSelf() with a Promise`".  Wordy, perhaps, but more explicit.

Reply:
Haha, I prefer it being more readable and explicit.